refactor(curriculum): JavaScript Objects Quiz (#58597)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/25-front-end-development/quiz-javascript-objects/66edcd0ecb4b25dc64a34804.md

`const { key: "defaultValue" } = object;`

#### --answer--

`const { key = "defaultValue" } = object;`

### --question--

#### --text--

How can you check if a `car` object has a `year` property?

#### --distractors--

`year.hasProperty("car");`

---

`car.hasProperty("year");`

---

`car.hasOwnProperty(year);`

#### --answer--

`car.hasOwnProperty("year");`

### --question--

#### --text--

What is an object method?

#### --distractors--

A string associated with an object.

---

An array associated with an object.

---

A number associated with an object.

#### --answer--

A function associated with an object.

### --question--

#### --text--

How can you call the `add` method of this `calculator` object to add the numbers `10` and `7`?

```js
const calculator = {
  add: function(a, b) {
    return a + b;
  }
};
```

#### --distractors--

`calculator#add(10, 7);`

---

`calculator.call(add(10, 7));`

---

`add.calculator(10, 7);`

#### --answer--

`calculator.add(10, 7);`

### --question--

#### --text--

Which operator removes a property from an object?

#### --distractors--

`remove`

---

`erase`

---

`discard`

#### --answer--

`delete`

### --question--

#### --text--

> How can you check if a `score` attribute exists in a `player` object?

Reviewer: ```suggestion
How can you check if a `score` property exists in a `player` object?
```

#### --text--

What is optional chaining used for in JavaScript?

#### --distractors--

To define variables with default values.

---

To define optional methods within an object that may or may not be called.

---

To write more concise code by chaining multiple method calls together in a single statement.

#### --answer--

> To access nested object properties without causing errors if a property is undefined or null.

Reviewer 2: I would remove the nested object part

Also, I would suggest that the distractors start with used to

```suggestion
Used to safely access properties of an object that might be `null` or `undefined`
```

Reviewer: ✅
